_plugins/tabs.rb

```diff
@@ -1,17 +1,46 @@
 #
-# This plugin generates bootstrap by providing a predefined block in Jekyll
+# Authour : Reza R <54559947+frozenprocess@users.noreply.github.com>
+# This plugin adds bootstrap predefined tab block in Jekyll
+<<-EXAMPLE
+You can generate tabs by using {% tabs type:pills %}
+@input id optional, html id
+You can link pane by their index position if you provide an integer for `tab-group` or 
+link panes using identical pane names
+@input tab-group optional, int|"name"
+@input type optional, pill|tabs
+To create individual tab use <name:Mytab,active:true>
+@input id optional, tab HTML id
+@input active optional, adds active class to tab
+@input name optional, tab visual name
+```
+   Normal tab                 |    Linked "Numerical index"    |   Linked tab "Identical names"                    
```

Comment:
I'm finding these examples slightly confusing.  The normal tab example only has a single tab in it, which doesn't sounds normal.  The identical names tabs all have the same name.

In addition, not sure if I understand the behavior.  Is it: if tab-group is a number then link tabs within the group tabs with the same tab-group number by tab index; if tab-group is name then link all tabs on the page by name, without ability to support different tag groups?

Reply:
That was an old example, hopefully new one can be useful. :)

Each tab (parent) can have as many panes (child) as user wants.
`Normal tab` generates a simple tab for user.
```
{% tab %}
<name:pane1,active:true>
<% 
this content belongs to `pane 1` and it will be selected by default since you declared `active:true`
in pane header.
  %>
<>
<% this content belongs to second pane with a randomly generated name %>
<name:This pane has no content>
<%%>
```
---
"Numerical index" implementation uses `tab-group` value to group multiple tabs together.
Selection process uses element index from DOM.
**Note: You are requried to provide `tab-group` with an integer to implement this type of tabs.**
```
{% tabs name:tab1,tab-group:1 %}             
<name:Non unique name,active:true> 
<% Content for tab1 %>             
<name:Non unique name>             
<% Content for tab2 %>             
<>                                 
<% Tab with random name  %>        
{% endtabs %}                      
{% tabs name:tab2,tab-group:1 %}             
<name:Non unique name,active:true> 
<% Content for tab1 %>             
<name:Non unique name>             
<% Content for tab2 %>             
<>                                 
<% Tab with random name  %>        
{% endtabs %}                  
{% tabs name:tab1,tab-group:2 %}             
<name:Non unique name,active:true> 
<% Content for tab1 %>             
<name:Non unique name>             
<% Content for tab2 %>             
<>                                 
<% Tab with random name  %>        
{% endtabs %}                      
{% tabs name:tab2,tab-group:2 %}             
<name:Non unique name,active:true> 
<% Content for tab1 %>             
<name:Non unique name>             
<% Content for tab2 %>             
<>                                 
<% Tab with random name  %>        
{% endtabs %}                       
```
---
"Identical names" implementation can be generated when `tab-group` is set to "name" at the parent level. This type of link happens between panes(child), using identical pane names user can link panes regardless of their position in DOM.
```
{% tabs tab-group:name %}      
<name:unique tab y,active:true>
<% Content for tab y %>        
<name:unique tab x>            
<% Content for tab x %>        
{% endtabs %}                  
{% tabs tab-group:name %}      
<name:unique tab x>
<% Content for tab x %>        
<name:unique tab y> 
<% Content for tab y %>        
{% endtabs %}                  
```

Reply:
@lxpollitt I think it would be better to change `name:` to `label:` in order to not confuse user with HTML name tag.
any thoughts?

Reply:
You can create tabs using a tab block `{% tabs %}`, Each tab (parent) can support as unlimited number of panes (child). These tabs are linked together by default, if you wish to separate tabs you can assign them to different groups using `tab-group` variable when creating the block `(ex. {% tabs tab-group:calico %}).

>**Link mechanism:** when multiple tabs belong to a `tab-group`, panes with identical labels will be linked together and as the result changing one pane effects other panes as well.

---

## Example
```
{% tabs %}
<label:firstpane>
<% First pane belongs to the default group %>
<label:secondpane>
<% Second pane belongs to the default group %>
{% endtabs %}
{% tabs %}
<label:firstpane>
<% First pane belongs to the default group %>
<label:secondpane>
<% Second pane belongs to the default group %>
{% endtabs %}

{% tabs tab-group=security %}
<label:firstpane>
<% First pane belongs to the security group %>
<label:secondpane>
<% First pane belongs to the security group %>
{% endtabs %}
{% tabs tab-group=security %}
<label:firstpane>
<% First pane belongs to the security group %>
<label:secondpane>
<% First pane belongs to the security group %>
{% endtabs %}
```